### PR TITLE
SUS-1387: fix namespace error in WallExternalController

### PR DESCRIPTION
**Pull request by mszabo-wikia**
ticket: https://wikia-inc.atlassian.net/browse/SUS-1387

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 28 issues
* MAJOR 7 major
* MINOR 14 minor
* INFO 7 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WallExternalController.class.php#L10](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L10): Class "WallExternalController" has 26 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WallExternalController.class.php#L275](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L275): The Cyclomatic Complexity of this function "deleteMessage" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. MAJOR [WallExternalController.class.php#L275](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L275): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallExternalController.class.php#L306](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L306): Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. MAJOR [WallExternalController.class.php#L337](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L337): Reduce this "switch/case" number of lines from 16 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. MAJOR [WallExternalController.class.php#L450](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L450): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. MAJOR [WallExternalController.class.php#L710](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L710): Remove this unused private "getDisplayName" method. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1144)
1. MINOR [WallExternalController.class.php#L56](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L56): Define a constant instead of duplicating this literal "value" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallExternalController.class.php#L88](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L88): Define a constant instead of duplicating this literal "error" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallExternalController.class.php#L126](https://github.com/Wikia/app/blob/438e6f8f6cf82654adea45be25f8900744b3c5d6/extensions/wikia/Wall/WallExternalController.class.php#L126): Define a constant instead of duplicating this literal "count" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/Wall/tests/WallExternalControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Unit test for WallExternalController, main AJAX entry point for Wall and Forum actions
+ */
+class WallExternalControllerTest extends WikiaBaseTest {
+	const TEST_EDIT_TOKEN = 'Egy aprócska kalapocska, benne csacska macska mocska.';
```
Comment (macbre) [inline extensions/wikia/Wall/tests/WallExternalControllerTest.php:7]: :hungary:  :)

File: extensions/wikia/Wall/WallExternalController.class.php [hunk 2]
```diff
@@ -230,13 +243,16 @@ public function postNewMessage() {
 			$titleMeta = $helper->getDefaultTitle();
 		}
 
-		if ( empty( $body ) ) {
+		$ns = $this->request->getInt( 'pagenamespace' );
+
+		// SUS-1387: Namespace parameter must be valid Wall or Forum namespace
+		if ( empty( $body ) || !in_array( $ns, $this->wg->WallNS ) ) {
```
Comment (macbre) [inline extensions/wikia/Wall/WallExternalController.class.php]: There's a `WallHelper::isWallNamespace( $ns )` helper method.

**Review by macbre — CHANGES_REQUESTED** [review]
See above

**Comment by WikiaTech** [comment]
SonarQube analysis reported 28 issues
* MAJOR 7 major
* MINOR 14 minor
* INFO 7 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WallExternalController.class.php#L10](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L10): Class "WallExternalController" has 26 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WallExternalController.class.php#L275](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L275): The Cyclomatic Complexity of this function "deleteMessage" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. MAJOR [WallExternalController.class.php#L275](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L275): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallExternalController.class.php#L306](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L306): Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. MAJOR [WallExternalController.class.php#L337](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L337): Reduce this "switch/case" number of lines from 16 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. MAJOR [WallExternalController.class.php#L450](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L450): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. MAJOR [WallExternalController.class.php#L710](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L710): Remove this unused private "getDisplayName" method. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1144)
1. MINOR [WallExternalController.class.php#L56](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L56): Define a constant instead of duplicating this literal "value" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallExternalController.class.php#L88](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L88): Define a constant instead of duplicating this literal "error" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallExternalController.class.php#L126](https://github.com/Wikia/app/blob/8d32cf1882f3595d0dde949646bc083504f09e72/extensions/wikia/Wall/WallExternalController.class.php#L126): Define a constant instead of duplicating this literal "count" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Review by macbre — APPROVED** [review]

**Comment by ghost** [comment]
Tested and accepted. :medal_sports:

**Pull request merged**